### PR TITLE
feat(providers): detect `VERCEL` and `VERCEL_ENV` 

### LINKED ISSUE #54: Detect `VERCEL_ENV` env  as vercel provider
Moving out from https://github.com/nuxt/image/pull/840

(i am not quite sure why it was used before)

### PR DESCRIPTION
**Pull request by NozomuIkuta**
### 🔗 Linked issue

Resolves #54.

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.
  - not applicable

### CODE REVIEW

File: src/providers.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ const providers: InternalProvider[] = [
   ["TEAMCITY", "TEAMCITY_VERSION"],
   ["TRAVIS"],
   ["VERCEL", "NOW_BUILDER"],
+  ["VERCEL", "VERCEL_ENV"],
```
Comment (NozomuIkuta) [inline src/providers.ts:92]: Should I add `["VERCEL"]` as well?
According to [their docs](https://vercel.com/docs/concepts/projects/environment-variables/system-environment-variables#system-environment-variables), `VERCEL` is a binary string (`0`/`1`) and `VERCEL_ENV` is a string representing the environment, both of which are described as:

> The [Environment](https://vercel.com/docs/concepts/projects/environment-variables#environments) that the app is deployed and running on.

I might want to add `{ "ci": false }` as well, don't I?

Comment (pi0) [reply src/providers.ts:92]: Yes, `VERCEL_ENV` seems to be a runtime flag indicator and therefore is not a CI.

Comment (NozomuIkuta) [reply src/providers.ts]: @pi0 

Thank you for the comment.
I added `VERCEL` environment variable and `ci: false` flag to them.

**Comment by pi0** [comment]
Thanks!

**Pull request merged**